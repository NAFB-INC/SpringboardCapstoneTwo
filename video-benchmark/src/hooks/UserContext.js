import React from "react";

//storage place for app-wide data
const UserContext = React.createContext();

export default UserContext;
