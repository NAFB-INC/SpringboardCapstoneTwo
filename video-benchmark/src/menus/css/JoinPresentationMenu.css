.JoinPresentationMenu{
    position:absolute;
    margin-top:5%;
    margin-left:5%;
    margin-right:5%;
    padding-top:2%;
    height:70%;
    width:88%;
    border:5px solid #6D2E46;
    background-color: #ECE2D0;
}

.found-video{
    font-size: 130%;
    overflow: auto;
    border-top: 10px solid #471e2e;
    border-bottom: 10px solid #471e2e;
    height:92%;
    padding-top:1%;
    font-weight:bolder;
}

.video-card{
    border:4px solid #6D2E46;
    margin-left:35%;
    margin-right:35%;
    background-color:#CEBEBE;
    padding:1%;
    margin-top:1%;
    font-weight: initial;
}

.video-card a{
    font-size:110%;
    text-decoration: none;
}

.video-card-sub{
    font-size:80%;
    font-style: italic;
}