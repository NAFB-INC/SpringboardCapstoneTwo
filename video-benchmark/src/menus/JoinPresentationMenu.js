import "./css/JoinPresentationMenu.css";
import React, { useContext,useState,useEffect } from "react";
import { Link } from "react-router-dom";
import UserContext from "../hooks/UserContext";
import CodeForm from "../forms/CodeForm";
import VideoAPI from "../api/VideoAPI";

function JoinPresentationMenu({myUser}) {
    const { stage,setStage } = useContext(UserContext);
    const [code,setCode] = useState("");
    const [video,setVideo] = useState(false);
    const [user,setUser] = useState("");


    useEffect(function getVideoByCode() {
        async function getVideo() {
            let myVideo = await VideoAPI.fetchVideo(code);
            console.log("fetched:",myVideo);
            if(myVideo["current"]){
                setVideo(myVideo["video"]);
                if(user==="presenter") {
                    sessionStorage.setItem('code',`${code}`);
                    sessionStorage.setItem('secure_hash','not_secure');
                }
            } else if(myVideo["past_video"]){
                setVideo(myVideo["video"]);
            }
        }
        if(code && stage > 2){
            getVideo();
        }
      }, [code,stage,setStage,user]);

    useEffect(function determineMyUser(){
        if(window.location.href.includes("presenter") && stage !== 3){
            setUser(myUser);
            setStage(1);
        }
        else if (window.location.href.includes("audience") && stage !== 3) {
            setUser(myUser);
            setStage(2);
        }
        console.log(window.location.href.includes("audience"))
    },[stage,setStage,setUser,myUser])

    function stageSelection(){
        if(stage===1){
            return(
                <div>
                    <CodeForm setCode={setCode} presenter={true}/>
                </div>
            );
        }
        if(stage===2){
            return(
                <div>
                    <CodeForm setCode={setCode}/>
                </div>
            );
        }
        else if(stage===3){
            if(!video){
                return(
                    <div>
                        Retrieving Video Details...
                    </div>
                );
            } else {
                return(
                    <div className="found-video">
                        Found a match!
                        <div className="video-card">
                            <div><Link to={`/${user}/${code}`}> {video.title} </Link> </div>
                            <div className="video-card-sub">presented by {video.presenter.first} {video.presenter.last}</div>
                            <div className="video-card-sub"><small>{video.duration}</small></div>
                        </div>
                        
                    </div>
                );
            }
        }else{
            setStage(2)
        }
    };
    
    return (
        <div className="JoinPresentationMenu">
            {stageSelection()}
        </div>
    );
}

export default JoinPresentationMenu;