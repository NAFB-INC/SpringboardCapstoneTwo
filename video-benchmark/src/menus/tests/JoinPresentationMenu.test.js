import React from "react";
import { render } from '@testing-library/react';
import JoinPresentationMenu from '../JoinPresentationMenu';
import UserContext from "../../hooks/UserContext";

let stage = 1;
let setStage = function(){};
const testCase1 = (
  <div>
      <UserContext.Provider value={{ stage,setStage }}>
          <JoinPresentationMenu />
      </UserContext.Provider>
  </div>);


it("renders without crashing", function() {
  render(testCase1);
});

it("matches snapshot", function() {
  const { asFragment } = render(testCase1);
  expect(asFragment()).toMatchSnapshot();
});