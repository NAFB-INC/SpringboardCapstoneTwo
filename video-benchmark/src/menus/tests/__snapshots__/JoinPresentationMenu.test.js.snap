// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="JoinPresentationMenu"
    >
      <div>
        <div
          class="AudienceCodeForm"
        >
          <div
            class="form-text"
          >
            Have a presentation code and password? Please enter it here!
          </div>
          <form>
            <div
              class="form-box"
            >
              <label
                for="audience_code"
              >
                Code: 
              </label>
              <input
                autocomplete="audience-code"
                name="audience_code"
                placeholder="abc123"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-box"
            >
              <label
                for="presenter_pass"
              >
                Password: 
              </label>
              <input
                autocomplete="presenter-password"
                name="presenter_pass"
                placeholder="xyzpdq"
                type="password"
                value=""
              />
            </div>
            <div />
            <button
              class="submit-btn"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
